data/constants/keycodes/keycodes_0.0.1.hjson

```diff
@@ -0,0 +1,94 @@
+{
+    "ranges": {
+        "0x0000/0x00FF": {
+            "define": "QK_BASIC"
+        },
+        "0x0100/0x1EFF": {
+            "define": "QK_MODS"
+        },
+        "0x2000/0x1FFF": {
+            "define": "QK_MOD_TAP"
+        },
+        "0x4000/0x0FFF": {
+            "define": "QK_LAYER_TAP"
+        },
+        "0x5000/0x001F": {
+            "define": "QK_TO"
+        },
+        "0x5020/0x001F": {
+            "define": "QK_MOMENTARY"
+        },
+        "0x5040/0x001F": {
+            "define": "QK_DEF_LAYER"
+        },
+        "0x5060/0x001F": {
+            "define": "QK_TOGGLE_LAYER"
+        },
+        "0x5080/0x001F": {
+            "define": "QK_ONE_SHOT_LAYER"
+        },
+        "0x50A0/0x001F": {
+            "define": "QK_LAYER_TAP_TOGGLE"
+        },
+        // 0x50C0/0x001F - UNUSED
+        // 0x50E0/0x001F - UNUSED
+        "0x5100/0x00FF": {
+            "define": "QK_ONE_SHOT_MOD"
+        },
```

Comment:
The current implementation of `OSM()` uses the same 5-bit `MOD_xxxx` encoding as many other features, therefore the size of this range can be reduced to `0x001F`.

Or do we want to extend just `OSM()` to support arbitrary 8-bit modifier masks? That would be rather confusing, because there would be no way to extend other keycodes to do the same. Also there are no free bits in the corresponding `action_t` encoding, unless we change it.

(We could make the `OSM()` macro translate the 5-bit encoding to 8-bit, and then add another macro to support the full 8-bit encoding, but I'm not sure that there would be any use case for those arbitrary right+left mod combinations.)

Reply:
Consistency would get my vote for the current refactoring pass, keeping the accepted mod values the same.